### ISSUE TITLE
Not working for derived methods and properties

### ISSUE THREAD

**Issue by kudlasek188** [HvyIndustries/crane#211]
Hi, does it have to work on dervied methods & properties? Because it's not working in my case.

**Comment by nevadascout** [comment]
Hey, thanks for the question!

Can you provide a code example of what is not working?

**Comment by ichiriac** [comment]
Hi @kudlasek188,

Actually class inheritance is supported by the Crane :

![image](https://cloud.githubusercontent.com/assets/173203/23696145/b8ef1fb4-03e2-11e7-993c-5ae7a555b632.png)

Here a code sample that works :

```php
<?php
class bar {
    public $barProperty = 'bar';
    public function doBar() {
        return $this->barProperty;
    }
}

class foo extends bar {
    public $fooProperty = 'foo';
    public function doFoo() {
        return $this->
    }
}
```

Can you check you are using the latest version of Crane (CTRL+MAJ+X and check Crane extension) and enable from settings the `crane.debugMode`.

We need a snippet code to reproduce the error, and if any, the debug ouput with error messages.

**Comment by kudlasek188** [comment]
Hi! @nevadascout sorry for replying this late. I guess problem is caused by the fact, that classes are in different files. It's quite strange, because after keyword "extends" it's suggesting name of parent class correctly, but it don't suggest its methods etc. from child's methods.

While testing this I found different bug. When are two classes in same namespace (for example Test), then after typing "extends" it's correcting it to Test\ParentClass instead of just ParentClass. It's okay, but then i need to change namespace at more places when it's changes. 

Example code:

```php
<?php

namespace Test;

class ParentClass {
    public function recreate() {
        return new ParentClass();
    }

    public function doThings(ParentClass $class) {
        return $class->foob();
    }
}

class ChildClass extends Test\ParentClass /* This suggestion is unoptimized */ {
    public function __construct() {
        $this->doThings(); // This suggestion works only when classes are in same file
    }
}
````

**Comment by nevadascout** [comment]
Thanks for replying!

I checked your code sample, and yes you are correct -- the `Test\` should not be added when extending a class in the same namespace.

I also checked and the reason that suggestions are not working is because of the `Test\` part in the extends section. If you remove this namespace part, then the suggestions start working correctly.

I've created two tasks to handle both these cases, which I will put into the next release for you.

Thanks again for helping us find (and solve) these problems! 😃
